js/private/js_library.bzl

```diff
@@ -68,48 +83,84 @@ target in {file_basename}'s package and add that target to the deps of {this_tar
 
     output_srcs = copy_files_to_bin_actions(ctx, ctx.files.srcs, is_windows = is_windows)
 
+    sources = []
+    declarations = []
+
     for src in output_srcs:
         if src.is_directory:
             # assume a directory contains typings since we can't know that it doesn't
-            typings.append(src)
+            declarations.append(src)
+            sources.append(src)
         elif (
             src.path.endswith(".d.ts") or
```

Comment:
I guess it's not really part of this PR but I think this also needs to include ".d.mts" and ".d.cjt" (https://github.com/aspect-build/rules_ts/commit/6a2841c4b6dd0760cc58d41e8762f5ab3833fb49)

Reply:
Oh right. There are more variants now.

Reply:
I put this function in a central helper that downstream rules will use as well. We can add the additional variants in a follow-up.

Reply:
Actually. I just updated here to save time.